feat(COC): add coc operation script order resource (#6841)
huaweicloud/terraform-provider-huaweicloud: docs/resources/coc_script_order_operation.md

---
subcategory: "Cloud Operations Center (COC)"
layout: "huaweicloud"
page_title: "HuaweiCloud: huaweicloud_coc_script_order_operation"
description: |-
  Use this resource to operation script order.
---

# huaweicloud_coc_script_order_operation

Use this resource to operation script order.

> 

Reviewer: ```suggestion

Use this resource to operation script order.

-> Deleting script order operation resource is not supported, it will only be removed from the state.

```

> ~> Deleting operate resource is not supported, it will only be removed from the state.

Reviewer: ```suggestion
~> Deleting script order operation resource is not supported, it will only be removed from the state.
```

## Example Usage

> ### Basic Usage

Reviewer 2: it is not needed, should be deleted

variable "execute_uuid" {}
variable "operation_type" {}

resource "huaweicloud_coc_script_order_operation" "test" {
  execute_uuid   = var.execute_uuid
  operation_type = var.operation_type
}
```

> ~> Deleting operate resource is not supported, it will only be removed from the state.

Reviewer: ```suggestion
```

## Argument Reference

The following arguments are supported:

* `execute_uuid` - (Required, String, NonUpdatable) Specifies the script order ID.

> * `batch_index` - (Optional, Int, NonUpdatable) Specifies the batch index.
> 
> * `instance_id` - (Optional, Int, NonUpdatable) Specifies the instance ID.
> 
> * `operation_type` - (Required, String, NonUpdatable) Specifies the operation type.
>   Values can be as follows:
>   + **CANCEL_INSTANCE**: Cancel an instance.
>   + **SKIP_BATCH**: Skip batch.
>   + **CANCEL_ORDER**: Cancel the script order.
>   + **PAUSE_ORDER**: Pause the script order.
>   + **CONTINUE_ORDER**: Continue the script order.

Reviewer: ```suggestion
* `operation_type` - (Required, String, NonUpdatable) Specifies the operation type.
  Values can be as follows:
  + **CANCEL_INSTANCE**: Cancel an instance.
  + **SKIP_BATCH**: Skip batch.
  + **CANCEL_ORDER**: Cancel the script order.
  + **PAUSE_ORDER**: Pause the script order.
  + **CONTINUE_ORDER**: Continue the script order.

* `batch_index` - (Optional, Int, NonUpdatable) Specifies the batch index.

* `instance_id` - (Optional, Int, NonUpdatable) Specifies the instance ID.
```
